c:geo crashes on resume in CacheDetailActivity

### Describe your problem!

c:geo crashes when CacheDetailActivity is resumed after low-memory situation.

### How to reproduce?

1. Open cache details
2. Put c:geo into background
3. Do something that uses lots of memory and causes the CacheDetailActivity to be suspended
4. Switch back to c:geo

### Actual result after these steps?

c:geo crashes with

```
12-04 19:13:52.760  8835  8835 W cgeo    : [main] UncaughtException
12-04 19:13:52.760  8835  8835 W cgeo    : java.lang.RuntimeException: Unable to start activity ComponentInfo{cgeo.geocaching/cgeo.geocaching.CacheDetailActivity}: java.lang.ClassCastException: android.view.AbsSavedState$1 cannot be cast to android.widget.ScrollView$SavedState
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2684)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.app.ActivityThread.-wrap12(ActivityThread.java)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.os.Handler.dispatchMessage(Handler.java:102)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.os.Looper.loop(Looper.java:154)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.app.ActivityThread.main(ActivityThread.java:6186)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at java.lang.reflect.Method.invoke(Native Method)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
12-04 19:13:52.760  8835  8835 W cgeo    : Caused by: java.lang.ClassCastException: android.view.AbsSavedState$1 cannot be cast to android.widget.ScrollView$SavedState
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.widget.ScrollView.onRestoreInstanceState(ScrollView.java:1832)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.view.View.dispatchRestoreInstanceState(View.java:15755)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:3231)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.view.View.restoreHierarchyState(View.java:15733)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.fragment.app.Fragment.restoreViewState(Fragment.java:639)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.fragment.app.Fragment.restoreViewState(Fragment.java:3010)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.fragment.app.Fragment.performActivityCreated(Fragment.java:3001)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.fragment.app.FragmentStateManager.activityCreated(FragmentStateManager.java:580)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:285)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:112)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1647)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3128)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:3072)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:251)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:502)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:246)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at cgeo.geocaching.activity.AbstractActivity.onStart(AbstractActivity.java:305)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1249)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.app.Activity.performStart(Activity.java:6701)
12-04 19:13:52.760  8835  8835 W cgeo    : 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2647)
```

### Expected result after these steps?

No crash.

### Reproducible

Yes

### c:geo Version

2021.12.04-NB-3b6e977

### System information

_No response_

### Additional Information

_No response_

--- a/main/res/layout/cache_filter_activity.xml
+++ b/main/res/layout/cache_filter_activity.xml
@@ -2,12 +2,12 @@
 <ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
-    android:id="@+id/activity_viewroot"
+    android:id="@+id/cachefilter_activity_viewroot"
     android:layout_width="fill_parent"
     android:layout_height="fill_parent"
     android:orientation="vertical"
     android:layout_marginTop="10dip"
-    tools:context=".filter.FilterActivity" >
+    tools:context=".filters.gui.GeocacheFilterActivity" >
 
     <LinearLayout
         android:layout_width="fill_parent"

--- a/main/res/layout/cachedetail_variables_page.xml
+++ b/main/res/layout/cachedetail_variables_page.xml
@@ -2,7 +2,7 @@
 <ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
-    android:id="@+id/activity_viewroot"
+    android:id="@+id/variable_page_viewroot"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:orientation="vertical"

--- a/main/src/cgeo/geocaching/filters/gui/GeocacheFilterActivity.java
+++ b/main/src/cgeo/geocaching/filters/gui/GeocacheFilterActivity.java
@@ -89,7 +89,7 @@ public class GeocacheFilterActivity extends AbstractActionBarActivity {
     public void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setThemeAndContentView(R.layout.cache_filter_activity);
-        binding = CacheFilterActivityBinding.bind(findViewById(R.id.activity_viewroot));
+        binding = CacheFilterActivityBinding.bind(findViewById(R.id.cachefilter_activity_viewroot));
 
         binding.filterPropsCheckboxes.removeAllViews();
         this.andOrFilterCheckbox = ViewUtils.addCheckboxItem(this, binding.filterPropsCheckboxes, TextParam.id(R.string.cache_filter_option_and_or), R.drawable.ic_menu_logic);

